### PR TITLE
Remove some working tests from issues.targets

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Cc @dotnet/ilc-contrib

### CODE REVIEW

**Comment by filipnavara** [comment]
Fixes #86561

**Comment by MichalStrehovsky** [comment]
> Fixes #86561

And dotnet/runtimelab#1136 :)

**Review by vitek-karas — APPROVED** [review]

File: src/tests/baseservices/threading/coverage/OSThreadId/OSThreadId.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public sealed class OSThreadId
         public static int TestEntryPoint()
         {
             // The property to be tested is internal.
-            Type runtimeThreadType = typeof(object).Assembly.GetType("System.Threading.Thread");
+            Type runtimeThreadType = Type.GetType("System.Threading.Thread");
```
Comment (VSadov) [inline src/tests/baseservices/threading/coverage/OSThreadId/OSThreadId.cs:20]: It is a bit weird that we have tests for `CurrentOSThreadId`. It is not the point of this PR, but I do not think the property should exist. 

Outside of Windows, the implementation is not obvious. Should this return the OS identifier for a thread, if exists - like `gettid()` on Linux or a more portable/useful pthread ID - `pthread_self()`? 
The native ID would be only useful for interop and the choice of ID will depend on what you are interoping with. 

It should not be a property on Thread. It is good that it is at least not public.

Comment (jkotas) [reply src/tests/baseservices/threading/coverage/OSThreadId/OSThreadId.cs:20]: This is "thread ID for diagnostics tracing".

The same point was discussed in the PR that added this property and the test: https://github.com/dotnet/coreclr/pull/19002#discussion_r203599753 , and more recently in https://github.com/dotnet/runtime/pull/86226#discussion_r1202871525

Comment (VSadov) [reply src/tests/baseservices/threading/coverage/OSThreadId/OSThreadId.cs:20]: I think `OS` part is confusing. Should be `DiagnosticID` or something like that.

If the purpose is to be an identity of threads that are not necessarily alive, an incrementing counter might work better.

Abstract `OS` id might be useful in some scenarios like the same OS thread attached to two runtimes and one wants to know that. `OS` id is typically unique only as long as the thread is alive though.

Comment (jkotas) [reply src/tests/baseservices/threading/coverage/OSThreadId/OSThreadId.cs:20]: IIRC, this thread ID has to match thread ID used by other diagnostic tools and libraries (e.g. tools and libraries that come with the OS, that are outside our control). We cannot just manufacture our own IDs here.

**Pull request merged**